### DISCUSSION TITLE
RAW query as include

### DISCUSSION THREAD

**Discussion by jonaaix** [spatie/laravel-query-builder#798] [Q&A]
How can I set a raw query as include?

I tried this test code, but could not get it working.

```php
class TestInclude implements IncludeInterface {
   public function __invoke(Builder|\Illuminate\Database\Eloquent\Builder $query, string $include) {
      return \DB::table('variations')->whereId(3788);
   }
}
```

`$this->addAllowedInclude(AllowedInclude::custom('test123', new TestInclude()));`

**Comment by joaovdiasb** [comment]
You need to chain $builder, like $builder->join('variations', 'variations.id', 'related_table.foreign_id') to include that table
